adcm_aio_client/core/actions/_objects.py

```diff
@@ -29,41 +51,96 @@ def name(self: Self) -> str:
+    async def run(self: Self) -> Job:
+        from adcm_aio_client.core.objects.cm import Job
 
-    @async_cached_property
-    async def _mapping_rule(self: Self) -> list[dict] | None:
-        return (await self._rich_data)["hostComponentMapRules"]
+        await self._ensure_rich_data()
+
+        data = {"isVerbose": self._verbose, "shouldBlockObject": self._blocking}
+        if self._has_mapping:
+            mapping = await self.mapping
+            data |= {"hostComponentMap": mapping._to_payload()}
+        if self._has_config:
+            config = await self.config
+            data |= {"configuration": config._to_payload()}
+
+        response = await self._requester.post(*self.get_own_path(), "run", data=data)
+        return Job(requester=self._requester, data=response.as_dict())
 
     @async_cached_property
     async def mapping(self: Self) -> ActionMapping:
-        mapping_change_allowed = await self._mapping_rule
-        if not mapping_change_allowed:
+        await self._ensure_rich_data()
+
+        if not self._has_mapping:
             message = f"Action {self.display_name} doesn't allow mapping changes"
-            raise NoMappingRulesForActionError(message)
+            raise NoMappingInActionError(message)
 
         cluster = await detect_cluster(owner=self._parent)
         mapping = await cluster.mapping
         entries = mapping.all()
 
         return ActionMapping(owner=self._parent, cluster=cluster, entries=entries)
 
-    def set_verbose(self: Self) -> Self:
-        self._verbose = True
-        return self
+    @async_cached_property
+    async def config(self: Self) -> ActionConfig:
+        await self._ensure_rich_data()
 
-    @async_cached_property  # TODO: Config class
-    async def config(self: Self) -> ...:
-        return (await self._rich_data)["configuration"]
+        if not self._has_config:
+            message = f"Action {self.display_name} doesn't allow config changes"
+            raise NoConfigInActionError(message)
 
-    @async_cached_property
-    async def _rich_data(self: Self) -> dict:
-        return (await self._requester.get(*self.get_own_path())).as_dict()
+        configuration = self._configuration
+        data = ConfigData.from_v2_response(data_in_v2_format=configuration)
+        schema = configuration["configSchema"]
+
+        return ActionConfig(schema=schema, config=data, parent=self)
 
+    @property
+    def _is_full_data_loaded(self: Self) -> bool:
+        return "hostComponentMapRules" in self._data
 
-class ActionsAccessor(NonPaginatedChildAccessor):
+    @property
+    def _has_mapping(self: Self) -> bool:
+        return bool(self._mapping_rule)
+
+    @property
+    def _has_config(self: Self) -> bool:
+        return bool(self._configuration)
+
+    @property
+    def _mapping_rule(self: Self) -> list[dict]:
+        try:
+            return self._data["hostComponentMapRules"]
+        except KeyError as e:
```

Comment:
Why not `_ensure_rich_data()` but `raise` here?

Reply:
Because I don't want this property to be async

I want public interface to be aware of data content, not each method in Action

Generally I don't see this property as necessary at all, but it's in requirements, so we'll keep it in here for now